backend/typescript/rest/givecloud.ts

```diff
@@ -33,6 +33,16 @@ givecloudRouter.post(
   body("supporter.last_name", "supporter last_name is required").exists(),
   async (req: Request, res: Response) => {
     try {
+      if (process.env.HMAC_SECRET_KEY) {
```

Comment:
Add an else statement here, where process.env.HMAC_SECRET_KEY is undefined. This case should throw an error that says "No HMAC secret key set"